### PR TITLE
Fix handling of runlabel IMAGE and NAME

### PR DESCRIPTION
**Pull request by rhatdan**
Fixes: https://github.com/containers/podman/issues/10192

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
@rhatdan , can we hold that back until https://github.com/containers/podman/pull/10147 is merged?  It's terribly close to completion.

**Comment by rhatdan** [comment]
Sure, but I doubt this conflicts.

**Comment by vrothberg** [comment]
> Sure, but I doubt this conflicts.

Thanks! I touched the runlabel code quite a bit and want to prevent further merge conflicts.

File: pkg/domain/infra/abi/containers_runlabel.go [hunk 1]
```diff
@@ -178,13 +178,22 @@ func generateRunlabelCommand(runlabel string, img *image.Image, args []string, o
...
 
 func replaceName(arg, name string) string {
+	if arg == "NAME" {
+		return name
+	}
+
 	newarg := strings.ReplaceAll(arg, "$NAME", name)
-	return strings.ReplaceAll(newarg, "${NAME}", name)
+	newarg = strings.ReplaceAll(newarg, "${NAME}", name)
+	return strings.ReplaceAll(newarg, "=NAME", fmt.Sprintf("=%s", name))
```
Comment (giuseppe) [inline pkg/domain/infra/abi/containers_runlabel.go:187]: could  `NAME` be the prefix of another variable `=NAMEFOO`?

Comment (rhatdan) [reply pkg/domain/infra/abi/containers_runlabel.go:187]: Theoretically yes.

Comment (rhatdan) [reply pkg/domain/infra/abi/containers_runlabel.go]: Fixed

**Review by giuseppe — APPROVED** [review]
LGTM

File: pkg/domain/infra/abi/containers_runlabel.go [hunk 2]
```diff
@@ -180,13 +180,28 @@ func generateRunlabelCommand(runlabel string, img *libimage.Image, inputName str
...
 
 func replaceName(arg, name string) string {
+	if arg == "NAME" {
+		return name
+	}
+
 	newarg := strings.ReplaceAll(arg, "$NAME", name)
-	return strings.ReplaceAll(newarg, "${NAME}", name)
+	newarg = strings.ReplaceAll(newarg, "${NAME}", name)
+	if strings.HasSuffix(newarg, "=NAME") {
```
Comment (TomSweeneyRedHat) [inline pkg/domain/infra/abi/containers_runlabel.go:189]: Is it possible to have a trailing space after NAME here?  If so, we probably want to trim it.

Comment (rhatdan) [reply pkg/domain/infra/abi/containers_runlabel.go:189]: No these are arguments, only if the user wrapped something with "foo=NAME " would that happen, and in that case I think we should ignore.

**Comment by TomSweeneyRedHat** [comment]
LGTM
The fedora 34 test failed for reasons unknown, I've restarted

**Pull request merged**